presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoMetadata.java

```diff
@@ -250,6 +252,25 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
                 localProperties.build());
     }
 
+    @Override
+    public Optional<LimitApplicationResult<ConnectorTableHandle>> applyLimit(ConnectorSession session, ConnectorTableHandle table, long limit)
+    {
+        MongoTableHandle handle = (MongoTableHandle) table;
+
+        // MongoDB doesn't support limit number greater than integer max
+        if (limit > Integer.MAX_VALUE) {
+            return Optional.empty();
+        }
+
+        if (handle.getLimit().isPresent() && handle.getLimit().getAsLong() <= limit) {
```

Comment:
The documentation in https://docs.mongodb.com/manual/reference/method/cursor.limit/#cursor.limit mentions that  

> A limit() value of 0 (i.e. .limit(0)) is equivalent to setting no limit.

Is this true for the API `iterable.limit(0)` being used here as well?

Generally speaking the engine should optimize the `LIMIT 0` clause, but the mongodb behavior seems counterintuitive to me. If that is indeed the case, then should we encode/document that special case here?

Reply:
That's true. Though the situation shouldn't be happen as you know, I added following logic and a test. 
```java
        // MongoDB cursor.limit(0) is equivalent to setting no limit
        if (limit == 0) {
            return Optional.empty();
        }
```

Reply:
Ideally Presto will replace if it encounters a limit node with count as 0. Anyway it is good to have this check